docs/content/concepts/assets/asset-checks/subsetting-asset-checks.mdx

```diff
@@ -0,0 +1,158 @@
+---
+title: " Subsetting asset checks | Dagster Docs"
+description: "Learn how to execute subsets of asset checks."
+---
+
+# Subsetting asset checks
+
+In some cases, you may only want to execute some of the [asset checks](/concepts/assets/asset-checks) defined in a <PyObject object="multi_asset" decorator /> or <PyObject object="multi_asset_check" decorator />. For example, you may want to materialize a <PyObject object="multi_asset" decorator /> without executing the checks or only execute a certain set of checks.
+
+In this guide, we'll show you a few approaches to subsetting asset checks in <PyObject object="multi_asset_check" decorator pluralize /> and <PyObject object="multi_asset" decorator pluralize />.
+
+---
+
+## Prerequisites
+
+Before continuing, you should be familiar with:
+
+- [Asset checks](/concepts/assets/asset-checks)
+- [Multi-assets](/concepts/assets/multi-assets)
+
+---
+
+## Subsetting checks in @multi_asset_checks
+
+Using the <PyObject object="multi_asset_check" decorator /> decorator's `specs` and `can_subset` arguments, you can execute a subset of checks in a single op.
+
+Inside the body of the function, we can use <PyObject object="AssetCheckExecutionContext" displayText="AssetCheckExecutionContext.selected_asset_check_keys" /> to identify which computations to run. We can also set the decorator's `can_subset` parameter to `True` to execute a subset of the asset checks that the computation contains.
+
+As we don't know in advance which checks will be executed, we explicitly `yield` each asset check result that we're expected to create:
+
+```python file=/concepts/assets/asset_checks/subset_multi_asset_check.py
+import random
+from typing import Iterable
+
+from dagster import (
+    AssetCheckExecutionContext,
+    AssetCheckResult,
+    AssetCheckSeverity,
+    AssetCheckSpec,
+    multi_asset_check,
+)
+
+
+@multi_asset_check(
+    specs=[
+        AssetCheckSpec(name="asset_check_one", asset="my_asset_one"),
+        AssetCheckSpec(name="asset_check_two", asset="my_asset_two"),
+    ],
+    can_subset=True,
+)
+def the_check(context: AssetCheckExecutionContext) -> Iterable[AssetCheckResult]:
+    for check_key in context.selected_asset_check_keys:
+        asset_key = check_key.asset_key
+
+        # TODO: populate metadata
```

Comment:
note TODO. I think the use of random.random() here is more confusing than helpful- can we replace it with something closer to the below snippet? I think it's also more clear if we remove the for loop

e.g.


```python
    if (
        AssetCheckKey(AssetKey("my_asset_one"), "asset_check_one")
        in context.selected_asset_check_keys
    ):
        yield AssetCheckResult(passed=True, metadata={"foo": "bar"}, name="asset_check_one")
    if (
        AssetCheckKey(AssetKey("my_asset_two"), "asset_check_two")
        in context.selected_asset_check_keys
    ):
        yield AssetCheckResult(passed=True, metadata={"foo": "bar"}, name="asset_check_two")
```

Reply:
Done!